conn.go

```diff
@@ -488,3 +500,59 @@ type TraceTag struct {
 	// Action - specifies an action, such as an INSERT or UPDATE operation, in a module
 	Action string
 }
+
+func (c *conn) openProxyConnection(ctx context.Context) error {
```

Comment:
ensureContextUser ?
The name should mirror the usage: this has to be called everywhere to change the connection according to the user requested in the context.

Reply:
renamed